interpreter/cling/lib/Interpreter/CIFactory.cpp

```diff
@@ -1662,7 +1680,10 @@ namespace {
     // adjusted per transaction in IncrementalParser::codeGenTransaction().
     CGOpts.setInlining(CodeGenOptions::NormalInlining);
 
-    // CGOpts.setDebugInfo(clang::CodeGenOptions::FullDebugInfo);
+    // Add debugging info when debugging or profiling
+    if (debuggingEnabled || profilingEnabled)
+      CGOpts.setDebugInfo(codegenoptions::FullDebugInfo);
```

Comment:
On all platforms? (This might be the thing to do, just asking because of the `#ifdef __linux__` above)

Reply:
The debugging feature should work everywhere GDB works, so no ifdefs here.